ingestor-default/src/test/scala/za/co/absa/hyperdrive/ingestor/implementation/mongodbutils/EmbeddedMongoDbSingleton.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.hyperdrive.ingestor.implementation.mongodbutils
+
+import de.flapdoodle.embed.mongo.config.{MongodConfigBuilder, Net, RuntimeConfigBuilder}
+import de.flapdoodle.embed.mongo.distribution.Version
+import de.flapdoodle.embed.mongo.{Command, MongodExecutable, MongodStarter}
+import de.flapdoodle.embed.process.config.io.ProcessOutput
+import de.flapdoodle.embed.process.runtime.Network
+import org.slf4j.LoggerFactory
```

Comment:
Interesting, we use slf4j api mostly, so that the actual logging framework could be replaced at runtime. Not sure if it can be done in Spark jobs though since Spark provides log4j implementation.

Reply:
It was one of the first things I tried but failed. I don't remember the reason anymore precisely, but yes it has to do with Spark.